commit 7bda0b56ef9f8d775eef4781bfbc1f4ad3d2ad18
Author: Tomer Cabouly <tcabouly@nvidia.com>
Date:   Sun May 10 06:08:06 2026 +0000

    issue: 4917201 Fix unused-but-set warnings in ring_simple/ring_tap
    
    Build with --enable-opt-log=no compiles ring_logfunc to ((void)0),
    leaving 'freed' / 'count' assigned but unread and breaking
    -Werror=unused-but-set-variable.
    
    - ring_simple.cpp::put_tx_buffers: NOT_IN_USE(freed) before logfunc
    - ring_tap.cpp::mem_buf_tx_release: NOT_IN_USE(freed) before logfunc
    - ring_tap.cpp::mem_buf_desc_return_single_to_owner_tx: drop dead
      'count' (incremented but never logged or returned)
    
    Signed-off-by: Tomer Cabouly <tcabouly@nvidia.com>

diff --git src/vma/dev/ring_simple.cpp src/vma/dev/ring_simple.cpp
index 9cb894527ef3..c08fd0383af9 100644
--- src/vma/dev/ring_simple.cpp
+++ src/vma/dev/ring_simple.cpp
@@ -833,6 +833,9 @@ int ring_simple::put_tx_buffers(mem_buf_desc_t* buff_list)
 		count++;
 		buff_list = next;
 	}
+
+	// to suppress warning in case ring_logfunc is compiled out
+	NOT_IN_USE(freed);
 	ring_logfunc("buf_list: %p count: %d freed: %d\n", buff_list, count, freed);
 
 	return_to_global_pool();
diff --git src/vma/dev/ring_tap.cpp src/vma/dev/ring_tap.cpp
index e4f56d69581f..9e4a4b957331 100644
--- src/vma/dev/ring_tap.cpp
+++ src/vma/dev/ring_tap.cpp
@@ -495,8 +495,6 @@ void ring_tap::mem_buf_desc_return_single_to_owner_tx(mem_buf_desc_t* p_mem_buf_
 {
 	auto_unlocker lock(m_lock_ring_tx);
 
-	int count = 0;
-
 	if (likely(p_mem_buf_desc)) {
 		//potential race, ref is protected here by ring_tx lock, and in dst_entry_tcp & sockinfo_tcp by tcp lock
 		if (likely(p_mem_buf_desc->lwip_pbuf.pbuf.ref))
@@ -508,7 +506,6 @@ void ring_tap::mem_buf_desc_return_single_to_owner_tx(mem_buf_desc_t* p_mem_buf_
 			p_mem_buf_desc->p_next_desc = NULL;
 			free_lwip_pbuf(&p_mem_buf_desc->lwip_pbuf);
 			m_tx_pool.push_back(p_mem_buf_desc);
-			count++;
 		}
 	}
 
@@ -547,6 +544,8 @@ int ring_tap::mem_buf_tx_release(mem_buf_desc_t* buff_list, bool b_accounting, b
 		count++;
 		buff_list = next;
 	}
+	// to suppress warning in case ring_logfunc is compiled out
+	NOT_IN_USE(freed);
 	ring_logfunc("buf_list: %p count: %d freed: %d\n", buff_list, count, freed);
 
 	return_to_global_pool();
